### PR TITLE
Add formatter

### PR DESCRIPTION
**Pull request by sebastianrakel**
On arch the compiler fails on missing formatter, this PR should provide them

### CODE REVIEW

File: src/terminal/InputGenerator.h [hunk 1]
```diff
@@ -496,4 +496,67 @@ struct formatter<terminal::MouseTransport>
...
+{
+    template <typename ParseContext>
+    constexpr auto parse(ParseContext& ctx)
+    {
+        return ctx.begin();
+    }
+    template <typename FormatContext>
+    auto format(terminal::Key _value, FormatContext& _ctx)
+    {
+      std::vector<std::string> key_values = {"F1","F2",
```
Comment (christianparpart) [inline src/terminal/InputGenerator.h:511]:
```suggestion
      static constexpr auto mappings = std::array{"F1","F2",
```

File: src/terminal/InputGenerator.h [hunk 2]
```diff
@@ -496,4 +496,67 @@ struct formatter<terminal::MouseTransport>
...
+	"Numpad_1",
+	"Numpad_2",
+	"Numpad_3",
+	"Numpad_4",
+	"Numpad_5",
+	"Numpad_6",
+	"Numpad_7",
+	"Numpad_8",
+	"Numpad_9"};
+      return format_to(_ctx.out(), key_values[unsigned(_value)]);
```
Comment (christianparpart) [inline src/terminal/InputGenerator.h:559]:
```suggestion
      return format_to(_ctx.out(), mappings.at(static_cast<size_t>(_value)));
```

File: src/terminal/InputGenerator.h [hunk 3]
```diff
@@ -496,4 +496,67 @@ struct formatter<terminal::MouseTransport>
...
+    template <typename ParseContext>
+    constexpr auto parse(ParseContext& ctx)
+    {
+        return ctx.begin();
+    }
+    template <typename FormatContext>
+    auto format(terminal::Key _value, FormatContext& _ctx)
+    {
+      std::vector<std::string> key_values = {"F1","F2",
+	"F3",
```
Comment (christianparpart) [inline src/terminal/InputGenerator.h:512]: You can try running clang-format over this file to fix the indentation here. 

The command on the CLI would be something like: `clang-format -i  ./src/terminal/InputGenerator.h`

**Review by christianparpart — CHANGES_REQUESTED** [review]
Looks good. But maybe you want to try avoid unnecessary memory allocations by avoiding to use `std::vector` and use `std::array` instead.

Using member `at(index)` over `operator[](index)` also ensures you do not accidentally overflow as the `at(index)` member function will throw if out of bounds (the other will not).

**Comment by christianparpart** [comment]
FYI: I just realized that these are due to fmtlib being too new on Arch Linux. I just ran into the same problem locally when upgrading the Ubuntu 21.10 to 22.04 (dev). Do you want to finish this? Then just go ahead an remove the "WIP" prefix when done // mark this PR as "ready for review". :)

**Comment by sebastianrakel** [comment]
> FYI: I just realized that these are due to fmtlib being too new on Arch Linux. I just ran into the same problem locally when upgrading the Ubuntu 21.10 to 22.04 (dev). Do you want to finish this? Then just go ahead an remove the "WIP" prefix when done // mark this PR as "ready for review". :)

I can try to fix the others but maybe i need some help

**Comment by christianparpart** [comment]
> > FYI: I just realized that these are due to fmtlib being too new on Arch Linux. I just ran into the same problem locally when upgrading the Ubuntu 21.10 to 22.04 (dev). Do you want to finish this? Then just go ahead an remove the "WIP" prefix when done // mark this PR as "ready for review". :)
> 
> I can try to fix the others but maybe i need some help

just ping us on Discord #contour-dev, or say up until which point you want to do it, and I can do the rest :)

**Review by christianparpart — APPROVED** [review]
LGTM. Many thanks @sebastianrakel

**Pull request merged**